File: src/main/java/com/kopyl/commit/IssueValidator.java
package com.kopyl.commit;

import com.atlassian.jira.rest.client.api.domain.Issue;

public class IssueValidator {
    private final Issue issue;

    public IssueValidator(Issue issue) {
        this.issue = issue;
    }

    public Boolean valid() {
        if (this.issue.getSummary() == null || this.issue.getSummary().isEmpty()) {
            return false;
        }

        if (this.issue.getComponents() == null || this.issue.getComponents().iterator().next() == null) {
            return false;
        }

        return true;
    }
}

File: src/main/kotlin/com/kopyl/commit/IssueValidator.kt
package com.kopyl.commit

import com.atlassian.jira.rest.client.api.domain.Issue

class IssueValidator(private val issue: Issue) {
    fun valid(): Boolean {
        // TODO return error message as well, eg: {valid: false, message: "Components missing"}
        if (issue.summary == null || issue.summary.isEmpty()) {
            return false
        }
        if (issue.components == null || issue.components.iterator().next() == null) {
            return false
        }

        return true
    }
}
